File: connect4-frontend/src/model/dto/GameDTO.ts
import type {GameBoardCellState} from "@/model/enum/GameBoarCellState.ts";

export interface GameDTO {
    lobbyId: string;
    state: 'RUNNING' | 'ENDED';
    currentTurn: string;
    player1: string;
    player2: string;
    board: {
        grid: GameBoardCellState[][];
    };
    turnResult: 'VALID' | 'DRAW' | 'WON';
    username: string;
}

File: connect4-gaming/src/main/java/de/thbingen/connect4/gaming/model/dto/GameDTO.java
package de.thbingen.connect4.gaming.model.dto;

import de.thbingen.connect4.common.model.enums.GameState;

public record GameDTO(String lobbyId,
                      GameState state,
                      String currentTurn,
                      String player1,
                      String player2,
                      de.thbingen.connect4.common.model.dto.GameBoardDTO board) {
}
